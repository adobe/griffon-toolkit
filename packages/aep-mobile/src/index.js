// @create-index

export { default as clientInfoAndroid } from './clientInfoAndroid.js';
export { default as clientInfoIOS } from './clientInfoIOS.js';
export { default as configUpdateCommand } from './configUpdateCommand.js';
export { default as fakeEventCommand } from './fakeEventCommand.js';
export { default as genericPlaces } from './genericPlaces.js';
export { default as logForwardCommand } from './logForwardCommand.js';
export { default as mobileEvent } from './mobileEvent.js';
export { default as placesEntry } from './placesEntry.js';
export { default as placesExit } from './placesExit.js';
export { default as receivePlaces } from './receivePlaces.js';
export { default as requestPlaces } from './requestPlaces.js';
export { default as screenshotCaptureCommand } from './screenshotCaptureCommand.js';
export { default as startEventForwardingCommand } from './startEventForwardingCommand.js';
export { default as startPlacesMonitor } from './startPlacesMonitor.js';

