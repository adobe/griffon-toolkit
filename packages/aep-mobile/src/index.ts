// created from 'create-ts-index'

export * from './types';
export * from './analyticsResponse';
export * from './clientInfoAndroid';
export * from './clientInfoIOS';
export * from './configUpdateCommand';
export * from './configuration';
export * from './configurationUpdate';
export * from './edgeErrorResponse';
export * from './edgeRequest';
export * from './edgeResponse';
export * from './fakeEventCommand';
export * from './genericPlaces';
export * from './genericTrack';
export * from './identityResponse';
export * from './lifecycleStart';
export * from './locationAuthStatus';
export * from './logForwardCommand';
export * from './mediaCreateTrackerRequest';
export * from './mediaSessionCreated';
export * from './mobileEvent';
export * from './placesEntry';
export * from './placesExit';
export * from './pushTrackingRequest';
export * from './receivePlaces';
export * from './requestPlaces';
export * from './screenshotCaptureCommand';
export * from './setPushIdentifier';
export * from './sharedState';
export * from './sharedStateConfig';
export * from './sharedStateVersions';
export * from './startEventForwardingCommand';
export * from './startPlacesMonitor';
export * from './trackAction';
export * from './trackMedia';
export * from './trackState';
