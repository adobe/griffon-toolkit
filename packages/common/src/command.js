/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import * as kit from '@adobe/griffon-toolkit';
import schema from '../schemas/command.json';

/**
 * Contains constants and functions for a Outgoing Command.
 *
 * The structure for a Outgoing Command is as follows:
 * ```
 * {
 *   clientId: <string>,
 *   payload: <object>,
 *   timestamp: <number>,
 *   type: 'control'
 *   vendor: <string>,
 * }
 * ```
 *
 * @namespace command
 */

/**
 * Paths for the keys on a Outgoing Command
 *
 * @enum {string}
 */
const path = {
  /** A unique id that differentiates clients from one another.<br />Path is `clientId`. */
  clientId: 'clientId',

  /** An object with custom data describing the event.<br />Path is `payload`. */
  payload: 'payload',

  /** When the command was triggered.<br />Path is `timestamp`. */
  timestamp: 'timestamp',

  /** The type of event. For commands it's always 'control'..<br />Path is `type`. */
  rootType: 'type',

  /** The vendor of the plugin to receive the command.<br />Path is `vendor`. */
  vendor: 'vendor'
};

/**
 * Describes the number of parents this object has based off schema references. When checking for matches for example, we want to
 * use a schema that is more specific over a more generic schema
 *
 * @constant
 */
const parentDepth = 0;

/**
 * A label that can be used when describing this object
 */
const label = 'Outgoing Command';

/**
 * A grouping for this object
 */
const group = 'command';

/**
 * The value for `rootType` for a Outgoing Command.
 *
 * Path is `type`.
 *
 * @constant
 */
const ROOT_TYPE = 'control';

/**
 * Retrieves a value from the object. You can provide either a path or an alias.
 *
 * @function
 * @param {string} alias Path or alias
 * @param {*} data Data to search
 * @returns {*}
 */
const get = (alias, data) => {
  const func = (data2) => kit.search(path[alias] || alias, data2);
  if (!data) { return func; }
  return func(data);
};

/**
 * Returns the `clientId` from the Outgoing Command.
 * This is the a unique id that differentiates clients from one another.
 *
 * Path is `clientId`.
 *
 * @function
 * @param {object} source The Outgoing Command instance
 * @returns {string}
 */
const getClientId = kit.search(path.clientId);

/**
 * Returns the `payload` from the Outgoing Command.
 * This is the an object with custom data describing the event.
 *
 * Path is `payload`.
 *
 * @function
 * @param {object} source The Outgoing Command instance
 * @returns {object}
 */
const getPayload = kit.search(path.payload);

/**
 * Returns the data using the specified path from the payload
 * of the Outgoing Command.
 *
 * @function
 * @param {...string} path key in object
 * @param {object} source The Outgoing Command instance
 * @returns {*}
 */
const getPayloadKey = kit.curry(
  (searchPath, source) => kit.search(`${path.payload}.${searchPath}`, source)
);

/**
 * Returns the `timestamp` from the Outgoing Command.
 * This is the when the command was triggered.
 *
 * Path is `timestamp`.
 *
 * @function
 * @param {object} source The Outgoing Command instance
 * @returns {number}
 */
const getTimestamp = kit.search(path.timestamp);

/**
 * Returns the `rootType` from the Outgoing Command.
 * This is the the type of event. For commands it's always 'control'..
 *
 * Path is `type`.
 *
 * @function
 * @param {object} source The Outgoing Command instance
 * @returns {string}
 */
const getRootType = kit.search(path.rootType);

/**
 * Returns the `vendor` from the Outgoing Command.
 * This is the the vendor of the plugin to receive the command.
 *
 * Path is `vendor`.
 *
 * @function
 * @param {object} source The Outgoing Command instance
 * @returns {string}
 */
const getVendor = kit.search(path.vendor);

/**
 * Matcher can be used to find matching Outgoing Command objects.
 *
 * @see kit.match
 * @constant
 */
const matcher = kit.combineAll([
  'timestamp',
  'type==`control`'
]);

/**
 * Tests the provided source against the matcher to see if it's Outgoing Command event.
 *
 * @function
 * @param {object} source The Outgoing Command instance
 * @returns {boolean}
 * @see kit.isMatch
 */
const isMatch = (source) => kit.isMatch(matcher, source);
/**
 * Generates a Outgoing Command with the const values set.
 * Can be useful in testing.
 * Can provide additional data by providing a flat object of paths and values.
 *
 * @function
 * @param {...Function} input Overrides
 * @returns {object}
 */
const make = (input) => kit.expandWithPaths(path, {
  rootType: 'control',
  ...input
});

/**
 * Generates a Outgoing Command with some default values set.
 * Can be useful in testing.
 * Can override defaults and provide additional data by providing a flat object
 * of paths and values.
 *
 * @function
 * @param {...Function} input Overrides
 * @returns {object}
 */
const mock = (input) => kit.expandWithPaths(path, {
  clientId: 'appleABC',
  timestamp: Date.parse('12 Jan 2020 07:23:17 GMT'),
  rootType: 'control',
  vendor: 'com.adobe.mobile.sdk',
  ...input
});

/* ADD CUSTOM CONTENT BELOW */

// additional exports should be added here:
const customExports = {};

/* END CUSTOM CONTENT */
/* The content below is autogenerated. Do not make any changes */

export default {
  path,
  mock,
  make,
  schema,
  get,
  ...customExports,
  getClientId,
  getPayload,
  getPayloadKey,
  getTimestamp,
  getRootType,
  getVendor,
  isMatch,
  matcher,
  ROOT_TYPE,
  label,
  group,
  parentDepth
};
