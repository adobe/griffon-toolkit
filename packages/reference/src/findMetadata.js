/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import { curry, forEach, forEachObjIndexed } from 'ramda';
import * as common from '@adobe/griffon-toolkit-common';
import * as aep from '@adobe/griffon-toolkit-aep-mobile';

const references = [common, aep];

const isProperSchemaSource = (source) => !!source.label && !!source.isMatch;

/**
 * Takes an object and checks it against all known schemas to find any matches.
 * It will return the requested metadata from the match with the highest
 * parentDepth.
 *
 * @function
 * @param {string} source The metadata to request
 * @param {object} source The object to match against
 * @returns {string} label
 */
export default curry((type, source) => {
  let matchDepth = -1;
  let match = '';

  forEach(
    forEachObjIndexed((file) => {
      if (isProperSchemaSource(file)
        && file.parentDepth > matchDepth
        && file.isMatch
        && file.isMatch(source)
      ) {
        match = file[type];
        matchDepth = file.parentDepth;
      }
    }),
    references
  );
  return match;
});
