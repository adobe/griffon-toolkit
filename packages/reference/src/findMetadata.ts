/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import { assoc, curry, forEach, forEachObjIndexed, reduce } from 'ramda';
import * as common from '@adobe/griffon-toolkit-common';
import * as aep from '@adobe/griffon-toolkit-aep-mobile';
import * as edge from '@adobe/griffon-toolkit-edge';
import * as pushServices from '@adobe/griffon-toolkit-push-services';

const references = [
  common,
  aep,
  edge,
  pushServices
];

const isProperSchemaSource = (source) => !!source.label && !!source.isMatch;

/**
 * Takes an object and checks it against all known schemas to find any matches.
 * It will return the requested metadata from the match with the highest
 * parentDepth.
 *
 * @function
 * @param {string[]} types The metadata to request
 * @param {object} source The object to match against
 * @returns {string} label
 */
export default curry((types, source): { [key: string]: any } => {
  let matchDepth = -1;
  let matches = {};

  forEach(
    forEachObjIndexed((file: any) => {
      if (isProperSchemaSource(file)
        && file.parentDepth > matchDepth
        && file.isMatch
        && file.isMatch(source)
      ) {
        matches = reduce(
          (acc, type: string) => assoc(type, type === 'file' ? file : file[type], acc),
          {},
          types
        );
        matchDepth = file.parentDepth;
      }
    }),
    references
  );
  return matches;
});
