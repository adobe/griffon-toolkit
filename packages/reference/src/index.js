// @create-index

export { default as classesForGroup } from './classesForGroup.js';
export { default as findGroup } from './findGroup.js';
export { default as findLabel } from './findLabel.js';
export { default as findMetadata } from './findMetadata.js';

