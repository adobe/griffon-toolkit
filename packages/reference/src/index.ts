// created from 'create-ts-index'

export * from './classesForGroup';
export * from './findGroup';
export * from './findLabel';
export * from './findMetadata';
