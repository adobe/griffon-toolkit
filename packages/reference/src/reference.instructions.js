/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

/**
 * NOTE: this file is auto-generated. Any changes will be clobbered.
 */

import * as kit from '@adobe/griffon-toolkit';

export default [
  {
    matcher: kit.combineAll([
      'timestamp',
      'type==\'control\''
    ]),
    mods: {
      staticLabel: 'Outgoing Command',
      customLabel: 'Outgoing Command',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'timestamp',
      'type'
    ]),
    mods: {
      staticLabel: 'Generic Event',
      customLabel: 'Generic Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.type==\'configUpdate\'',
      'vendor==\'com.adobe.griffon.mobile\'',
      'timestamp',
      'type==\'control\''
    ]),
    mods: {
      staticLabel: 'Config Update Command',
      customLabel: 'Config Update Command',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.type==\'fakeEvent\'',
      'vendor==\'com.adobe.griffon.mobile\'',
      'timestamp',
      'type==\'control\''
    ]),
    mods: {
      staticLabel: 'Fake Event Command',
      customLabel: 'Fake Event Command',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.type==\'logForwarding\'',
      'vendor==\'com.adobe.griffon.mobile\'',
      'timestamp',
      'type==\'control\''
    ]),
    mods: {
      staticLabel: 'Log Forward Command',
      customLabel: 'Log Forward Command',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource',
      'payload.ACPExtensionEventType',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Generic Mobile Event',
      customLabel: 'Generic Mobile Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.type==\'screenshot\'',
      'vendor==\'com.adobe.griffon.mobile\'',
      'timestamp',
      'type==\'control\''
    ]),
    mods: {
      staticLabel: 'Screenshot Capture Command',
      customLabel: 'Screenshot Capture Command',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.type==\'startEventForwarding\'',
      'vendor==\'com.adobe.griffon.mobile\'',
      'timestamp',
      'type==\'control\''
    ]),
    mods: {
      staticLabel: 'Start Event Forwarding Command',
      customLabel: 'Start Event Forwarding Command',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.type==\'connect\'',
      'type==\'client\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Client Info Event',
      customLabel: 'Client Info Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'type==\'log\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Log Event',
      customLabel: 'Log Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'type==\'blob\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Screenshot Response Event',
      customLabel: 'Screenshot Response Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.responsecontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.analytics\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Analytics Response',
      customLabel: 'Analytics Response',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.deviceInfo."Canonical platform name"==\'Android\'',
      'payload.type==\'connect\'',
      'type==\'client\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'iOS Client Info Event',
      customLabel: 'iOS Client Info Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.deviceInfo."Canonical platform name"==\'iOS\'',
      'payload.type==\'connect\'',
      'type==\'client\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'iOS Client Info Event',
      customLabel: 'iOS Client Info Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.responsecontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.configuration\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Configuration Event',
      customLabel: 'Configuration Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.requestcontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.configuration\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Configuration Update',
      customLabel: 'Configuration Update',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.places\'',
      'payload.ACPExtensionEventSource',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Generic Places Event',
      customLabel: 'Generic Places Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.requestcontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.generic.track\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Generic Track Event',
      customLabel: 'Generic Track Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.responsecontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.lifecycle\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Lifecycle Start',
      customLabel: 'Lifecycle Start',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.sharedstate\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.hub\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Shared State Event',
      customLabel: 'Shared State Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.requestcontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventType.placesMonitor\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Start Monitor Event',
      customLabel: 'Start Monitor Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventData.requesttype==\'requestsetauthorizationstatus\'',
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.requestcontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.places\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Location Auth Status',
      customLabel: 'Location Auth Status',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventData.regioneventtype==\'entry\'',
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.responsecontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.places\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Places Entry Event',
      customLabel: (data) => `POI Entry: ${kit.search('payload.ACPExtensionEventData.triggeringregion.regionname', data)}`,
      customShortDescription: (data) => {
        const name = kit.search('payload.ACPExtensionEventData.triggeringregion.regionname', data);
        const street = kit.search('payload.ACPExtensionEventData.triggeringregion.regionmetadata.street', data);
        const city = kit.search('payload.ACPExtensionEventData.triggeringregion.regionmetadata.city', data);
        const address = street && city ? `${street}, ${city}` : street || city;
        const output = name && address ? `${name} - ${address}` : name || address;
        return `POI: ${output}`;
      }
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventData.regioneventtype==\'exit\'',
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.responsecontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.places\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Places Exit Event',
      customLabel: 'Places Exit Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventData.nearbypois',
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.responsecontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.places\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Receive Places Event',
      customLabel: 'Receive Places Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventData.requesttype==\'requestgetnearbyplaces\'',
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.requestcontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.places\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Request Places Event',
      customLabel: 'Request Places Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.sharedstate\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.hub\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Shared State - Versions',
      customLabel: 'Shared State - Versions',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.requestcontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.generic.track\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Track Action Event',
      customLabel: 'Track Action Event',
      customShortDescription: () => ''
    }
  },
  {
    matcher: kit.combineAll([
      'payload.ACPExtensionEventSource==\'com.adobe.eventsource.requestcontent\'',
      'payload.ACPExtensionEventType==\'com.adobe.eventtype.generic.track\'',
      'type==\'generic\'',
      'timestamp'
    ]),
    mods: {
      staticLabel: 'Track State Event',
      customLabel: 'Track State Event',
      customShortDescription: () => ''
    }
  }
];