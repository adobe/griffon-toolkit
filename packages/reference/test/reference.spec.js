/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import { findLabel, findGroup, classesForGroup } from '../src';
import entry from '../../aep-mobile/src/placesEntry';
import aep from '../../aep-mobile/src/mobileEvent';
import event from '../../common/src/event';
import command from '../../common/src/command';

const mockEntry = entry.mock();
const mockAEP = aep.mock();
const mockEvent = event.mock();
const mockCommand = command.mock();

describe('findLabel', () => {
  it('gets a label for the provided event', () => {
    expect(findLabel(mockEntry)).toBe('Places Entry Event');
    expect(findLabel(mockAEP)).toBe('Generic Mobile Event');
    expect(findLabel(mockEvent)).toBe('Generic Event');
    expect(findLabel({})).toBe('');
  });
});

describe('findGroup', () => {
  it('gets a group for the provided data', () => {
    expect(findGroup(mockEntry)).toBe('event');
    expect(findGroup(mockCommand)).toBe('command');
    expect(findGroup({})).toBe('');
  });
});

describe('classesForGroup', () => {
  it('retrieves all the classes for a group', () => {
    expect(Object.keys(classesForGroup('construct')).indexOf('annotation')).not.toBe(-1);
  });
});

