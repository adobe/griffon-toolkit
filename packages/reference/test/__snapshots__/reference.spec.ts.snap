// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classesForGroup retrieves all the classes for a group 1`] = `"payload.ACPExtensionEventData.regioneventtype==\`entry\` && (payload.ACPExtensionEventSource==\`com.adobe.eventSource.responseContent\` || payload.ACPExtensionEventSource==\`com.adobe.eventsource.responsecontent\`) && (payload.ACPExtensionEventType==\`com.adobe.eventType.places\` || payload.ACPExtensionEventType==\`com.adobe.eventtype.places\`) && timestamp"`;
